src/amp-story-player/amp-story-player-impl.js

```diff
@@ -833,29 +856,48 @@ export class AmpStoryPlayer {
    * Gets encoded url for player usage.
    * @param {string} href
    * @param {VisibilityState=} visibilityState
+   * @param {?EmbedMode=} embedMode If not provided, it will keep
+   *    the value provided by the user on the story URL. If not
+   *    defined, the value EmbedMode.NOT_EMBEDDED will be used.
    * @return {!Location}
    * @private
    */
-  getEncodedLocation_(href, visibilityState = VisibilityState.INACTIVE) {
-    const params = dict({
-      'amp_js_v': '0.1',
+  getEncodedLocation_(
+    href,
+    visibilityState = VisibilityState.INACTIVE,
+    embedMode
+  ) {
+    const noFragmentUrl = removeFragment(href);
+    const originalFragmentString = getFragment(href);
+    const {
+      [EmbedModeParam]: originalEmbedMode, // Original embedMode is treated separately so we don't override it
+      ...originalFragments
+    } = parseQueryString(originalFragmentString);
+
+    const newFragments = {
       'visibilityState': visibilityState,
       'origin': this.win_.origin,
       'showStoryUrlInfo': '0',
       'storyPlayer': 'v0',
       'cap': 'swipe',
+      [EmbedModeParam]:
+        embedMode ?? originalEmbedMode ?? EmbedMode.NOT_EMBEDDED,
```

Comment:
Referencing my comment above, this could be something like:

```
const newFragments = { ... }
const currentEmbedMode = embedModeOverride || originalEmbedMode;
if (currentEmbedMode) { newFragments[embedMode] = currentEmbedMode }

Reply:
I originally thought about this, but since `undefined` parameters are skipped in `serializeQueryString`, we could also only remove the last `?? EmbedMode.NOT_EMBEDDED`. Would it be better to be more explicit?